### ISSUE TITLE
Add support for determining mutual exclusivity among phenotypes

### ISSUE THREAD

**Issue by johnbradley** [phenoscape/rphenoscape#237]
Add support for determining whether phenotypes are mutually exclusive.
Initial code changes have been done in PR #30.

**Comment by johnbradley** [comment]
The initial code consists of two functions: mutual_exclusivity_pair() and mutual_exclusivity().

### [mutual_exclusivity_pair(phenotype.a, phenotype.b, ...)](https://github.com/phenoscape/rphenoscape/blob/ea9df50dbb73a0a78e90127aa834b24b4a7ef42c/R/mutual_exclusivity.R#L95)
Given two phenotypes determine their exclusivity ("strong_compatibility", "weak_compatibility", etc)

Example:
```
> phens <- get_phenotypes("basihyal bone", taxon = "Cyprinidae")$id
> mutual_exclusivity_pair(phens[1], phens[2])
[1] strong_exclusivity
5 Levels: strong_compatibility < weak_compatibility < inconclusive_evidence < ... < strong_exclusivity
```

### [mutual_exclusivity(phenotypes, ..)](https://github.com/phenoscape/rphenoscape/blob/ea9df50dbb73a0a78e90127aa834b24b4a7ef42c/R/mutual_exclusivity.R#L270)
Given a list of phenotypes return a list with `matrix` and `dataframe` keys that are derived from exclusivity values.
The `matrix` item is a matrix containing numeric representations of the factor levels returned by mutual_exclusivity_pair.
The `dataframe`- contains a row for each pair combination of input phenotypes.
The data frame columns are `id.1`, `label.1`, `id.2`, `label.2`, and `mutual-exclusivity`
 
Example:
```
> phens <- get_phenotypes("basihyal bone", taxon = "Cyprinidae")$id
> result <- mutual_exclusivity(c(phens[1], phens[2]))
> result$matrix
  1 2 3
1 1 5 4
2 5 1 2
3 4 2 1
> result$dataframe[c("label.1", "label.2", "mutual_exclusivity")]
                         label.1                        label.2 mutual_exclusivity
1 basihyal bone decreased length basihyal bone increased length strong_exclusivity
2 basihyal bone decreased length   basihyal bone increased size   weak_exclusivity
3 basihyal bone increased length   basihyal bone increased size weak_compatibility
```

**Comment by johnbradley** [comment]
I am wondering if we need to export both functions. 
Using `mutual_exclusivity()` you can recreate the behavior of `mutual_exclusivity_pair()` like so:
```
mutual_exclusivity(c(phen1, phen2))$dataframe$mutual_exclusivity
```

**Comment by johnbradley** [comment]
The [two examples for mutual_exclusivity()](https://github.com/phenoscape/rphenoscape/blob/ea9df50dbb73a0a78e90127aa834b24b4a7ef42c/R/mutual_exclusivity.R#L232-L268) are really slow. 
The first example is basically:
```
studies <- get_studies()
study <- studies$id[studies$label == 'Dillman et al. (2016)']
phenotypes <- get_phenotypes(study = study)
phenotypes_ids <- phenotypes$id
exclusivity <- mutual_exclusivity(phenotypes_ids)
```
The `mutual_exclusivity()` call takes over 14 hours to finish.
I haven't run the second example all the way through, but it looks like it would take a similar amount of time.

I'm wondering if there is a real world example we could give that finishes in a reasonable amount of time.

**Comment by johnbradley** [comment]
In branch [130-mutual-exclusion](https://github.com/phenoscape/rphenoscape/tree/130-mutual-exclusion) I made changes to use `mutually_exclusive` as the name for the function. 

I just noticed that there is a [mutual-exclusion.Rmd vignette](https://github.com/phenoscape/rphenoscape/blob/1be1a8a9206b16b721f27b5686677f4e4aa714ad/vignettes/mutual-exclusion/mutual-exclusion.Rmd#L327-L344). The `mutual-exclusion.Rmd` vignette contains examples that run much faster. I am going to update the documentation examples based on the vignette. 

I'm wondering if we should keep the `mutual-exclusion.Rmd` vignette. It does have a good bit more commentary than typical examples.
